### PR TITLE
HDDS-4076. Translate CSI.md into Chinese

### PR DESCRIPTION
**Pull request by maobaolong**
## What changes were proposed in this pull request?

Translate CSI.md into Chinese

## What is the link to the Apache JIRA

HDDS-4076

## How was this patch tested?

No need

### CODE REVIEW

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+获取更多CSI的信息，可以参考[SCI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md)
+
+Ozone CSI是 CSI 的一种实现，它使得在容器里使用Ozone成为了可能。 
+
+## 入门
+
+首先，我们需要一个带有S3Gateway的Ozone集群，并且它的OM和S3Gateway的端口都可以对CSI pod可见，
+因为CSIServer 将会访问 OM 来创建或者删除篮子，并且，CSIServer将会通过goofys创建到s3g的挂载点的方式来发布桶。 
+
+如果你没有一个运行在kubernetes上的Ozone集群，你可以参考[Kubernetes]({{< ref "start/Kubernetes.zh.md" >}}) 来创建一个。
```
Comment (cxorm) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:35]:
```suggestion
如果你没有一个运行在kubernetes上的Ozone集群，你可以参考[Kubernetes](https://github.com/apache/hadoop-ozone/blob/master/hadoop-hdds/docs/content/start/Kubernetes.zh.md) 来创建一个。
```

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## 创建 pv-test 并查看结果
+
+通过执行以下命令，创建 pv-test 相关的资源。
+
+```bash
+kubectl create -f /ozone/kubernetes/examples/ozone/pv-test
+```
+
+连接 pod scm-0 并且 创建一个键到篮子/s3v/pvc* 。
```
Comment (cxorm) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:53]:
```suggestion
连接 pod scm-0 并且 创建一个键到桶/s3v/pvc* 。
```

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 1 above)
Comment (adoroszlai) [reply hadoop-hdds/docs/content/interface/CSI.zh.md:35]: @cxorm I don't think we should hard-code URLs like that.  What's the problem with the Hugo `ref` shortcode?

**Comment by adoroszlai** [comment]
> ## How was this patch tested?
> No need

Although this is only a documentation change, I don't agree that tests can be skipped.  Some ideas for testing:

1. Use `mvn -pl :hadoop-hdds-docs clean package` to generate static docs.  It should succeed without errors.  Compile errors may be caused eg. by invalid syntax.
2. Open the generated page in a browser and
   * check that layout is fine (compare to English version).
   * check that there are no broken images (compare to English version).
   * verify all links work.

Hope that helps.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 1 above)
Comment (cxorm) [reply hadoop-hdds/docs/content/interface/CSI.zh.md:35]: Thank you @adoroszlai for the reminder.

Yeah the URL here could be not hard-code by Hugo. (I just show it in other .md editor and not consider Hugo)

No problem here.

Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md:35]: @cxorm https://ci-hadoop.apache.org/view/Hadoop%20Ozone/job/ozone-doc-master/lastSuccessfulBuild/artifact/hadoop-hdds/docs/public/interface/csi.html

**Comment by maobaolong** [comment]
@adoroszlai Thanks for tell me how to test the docs changes locally.

It works good, the link can works also.
image

PTAL.

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+容器存储接口 `Container Storage Interface` (CSI) 使存储供应商（SP）能够一次性开发一个插件，并让它跨多个容器编排工作。
+
+获取更多CSI的信息，可以参考[SCI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md)
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:26]: How about add one space before/after the English words. `多CSI的` -> `多 CSI 的`
Would you please keep above style consistent in the whole article?

Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Thanks @runitao for your suggestion, it make this document much pretty. Now, i addressed your comment, PTAL.

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+---
+title: CSI 协议
+weight: 3
+summary: Ozone 支持 容器存储接口(CSI) 协议。你可以通过Ozone CSI 挂载 Ozone 桶的方式使用 Ozone。
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:4]: `接口(CSI) 协` -> `接口 (CSI) 协`
`通过Ozone CSI 挂` -> `通过 Ozone CSI 挂`

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+容器存储接口 `Container Storage Interface` (CSI) 使存储供应商（SP）能够一次性开发一个插件，并让它跨多个容器编排工作。
+
+获取更多 CSI 的信息，可以参考[SCI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md)
+
+Ozone CSI 是 CSI 的一种实现，它可以将 Ozone 用作容器的存储卷。 
+
+## 入门
+
+首先，我们需要一个带有 S3Gateway 的 Ozone 集群，并且它的 OM 和 S3Gateway 的端口都可以对 CSI pod 可见，
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:32]: In the original English doc, the `s3gateway` is used.

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+因为 CSIServer 将会访问 OM 来创建或者删除篮子，并且，CSIServer 将会通过 goofys 创建到 s3g 的挂载点的方式来发布桶。
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:33]: bucket shoud be translated to `桶`
I think `同时 CSIServer 将` is better than `并且，CSIServer 将`.

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+Ozone CSI 是 CSI 的一种实现，它可以将 Ozone 用作容器的存储卷。 
+
+## 入门
+
+首先，我们需要一个带有 S3Gateway 的 Ozone 集群，并且它的 OM 和 S3Gateway 的端口都可以对 CSI pod 可见，
+因为 CSIServer 将会访问 OM 来创建或者删除篮子，并且，CSIServer 将会通过 goofys 创建到 s3g 的挂载点的方式来发布桶。 
+
+如果你没有一个运行在 kubernetes 上的 Ozone 集群，你可以参考[Kubernetes]({{< ref "start/Kubernetes.zh.md" >}}) 来创建一个。
+使用来自 `kubernetes/examples/ozone`的资源，你可以找到所有需要的kubernetes资源来和指定的CSI运行在一起
+(参考 `kubernetes/examples/ozone/csi`)
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:37]: miss spaces around kubernetes and CSI

I think we shoud use K in Kubernetes, although `kubernetes` is sometimes used in original doc.

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+## 创建 pv-test 并查看结果
+
+通过执行以下命令，创建 pv-test 相关的资源。
+
+```bash
+kubectl create -f /ozone/kubernetes/examples/ozone/pv-test
+```
+
+连接 pod scm-0 并且 创建一个键到桶 /s3v/pvc* 。
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:53]: `并在 /s3v/pvc* 桶创建一个键值。`

File: hadoop-hdds/docs/content/interface/CSI.zh.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  "name" : "pvc-861e2d8b-2232-4cd1-b43c-c0c26697ab6b",
+  "storageType" : "DISK",
+  "versioning" : false,
+  "creationTime" : "2020-06-11T08:19:47.469Z",
+  "encryptionKeyName" : null
+}
+[hadoop@scm-0 ~]$ ozone sh key put /s3v/pvc-861e2d8b-2232-4cd1-b43c-c0c26697ab6b/A LICENSE.txt
+```
+
+现在，通过映射 `ozone-csi-test-webserver-7cbdc5d65c-h5mnn` 的端口来通过浏览器看到 web 页面.
```
Comment (runitao) [inline hadoop-hdds/docs/content/interface/CSI.zh.md:70]: `端口来通过浏览器看到 web 页面.` -> `端口，我们可以使用浏览器展示其 UI 页面。`

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 1 above)
Comment (cxorm) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Thank you @maobaolong for the reply.
No problem here, it's my carelessness.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 4 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Done.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 5 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Done. I should keep consistent with the English version, thanks, I've corrected it.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 6 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Done.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 7 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md:37]: Done. Thanks for point it for me.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 8 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Done.

File: hadoop-hdds/docs/content/interface/CSI.zh.md (hunk 9 above)
Comment (maobaolong) [reply hadoop-hdds/docs/content/interface/CSI.zh.md]: Done.

**Comment by maobaolong** [comment]
@runitao Thanks advance for correct me so much translate issue, please take another look at the @runitao Thanks advance for correct me so much translate issue, please take another look at this PR.

**Comment by maobaolong** [comment]
@runitao Thanks for you review and improve my Chinese 😄
@cxorm @adoroszlai Thanks for the previous review, PTAL.

**Comment by cxorm** [comment]
LGTM. +1.
Thanks @runitao for the review, I think we could add some rules about translation on jira or wiki (such as "one space before/after the English words"), and anyone could refer the rules when contributing translation work.

**Comment by maobaolong** [comment]
@cxorm Something like this
- https://docs.alluxio.io/os/user/stable/en/contributor/Documentation-Conventions.html
- https://github.com/Alluxio/alluxio/wiki/Translation-Terms

**Comment by adoroszlai** [comment]
Thanks @maobaolong for the translation, and @cxorm and @runitao for the review.

**Pull request merged**